pkg/controller/daemon/daemoncontroller.go

```diff
@@ -401,14 +404,21 @@ func (dsc *DaemonSetsController) updateNode(old, cur interface{}) {
 		glog.V(4).Infof("Error enqueueing daemon sets: %v", err)
 		return
 	}
+	// TODO: it'd be nice to pass a hint with these enqueues, so that each ds would only examine the added node (unless it has other work to do, too).
 	for i := range dsList.Items {
 		ds := &dsList.Items[i]
-		shouldEnqueue := (dsc.nodeShouldRunDaemonPod(oldNode, ds) != dsc.nodeShouldRunDaemonPod(curNode, ds))
-		if shouldEnqueue {
+		_, oc1, oc2, err := dsc.nodeShouldRunDaemonPod(oldNode, ds)
```

Comment:
It's not clear what the names mean. Maybe `oldShouldSchedule, oldShouldContinueRunning`? Same for the names below.